### PR TITLE
Introduce 'Delta' data type to simplify diff checks

### PR DESCRIPTION
**Pull request by chshersh**
Minor refactoring that introduces our own `Delta` data type that simplifies filtering of diffs.

### CODE REVIEW

File: src/Hintman/Core/Delta.hs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{- | This module contains subset of all possible diff deltas that can
+be handled by Hintman. We consider for hints only the following deltas:
```
Comment (popzxc) [inline src/Hintman/Core/Delta.hs]: Nit: "We consider for hints only the following deltas" sounds a bit awkward.
I'd suggest something like "Hintman processes the deltas that satisfy the following criteria".

**Comment by chshersh** [comment]
@popzxc I think it's too early to copy-paste the code from the `diff-parse` package. It works fine an so far there were no problems with it. Instead of copy-pasting the package, we can fork it and use the patched version from GitHub. Or we can even take it over by following this procedure:

* [Taking over a package](http://wiki.haskell.org/Taking_over_a_package)

**Comment by popzxc** [comment]
@chshersh I didn't mean copy-pasting code from library itself, I only meant encapsulate the *usage* of it. So, the binding between `Text.Diff.Parse` and `Hintman` will be stored in one place, and if we will decide to do something with it - we will have to introduce changes only in one place.

Regarding taking over the package - yeah, that's a good option :)

File: src/Hintman/Core/Delta.hs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import Text.Diff.Parse.Types (Content (..), FileDelta (..), FileStatus (..), Hunk)
+
+
+{- | Wrapper around relevant fields of 'FileDelta'.
+-}
+data Delta = Delta
+    { deltaDestFile :: !Text
+    , deltaHunks    :: ![Hunk]
+    } deriving (Show)
```
Comment (vrom911) [inline src/Hintman/Core/Delta.hs]: This is going to be our default standard soon anyway :)
```suggestion
    } deriving stock (Show)
```

**Pull request merged**